### PR TITLE
Profiler Keyboard/Mouse UX changes to align with Systrace.

### PR DESCRIPTION
**Pull request by pmuetschard**
- w/s for zooming instead of q/e.
- zoom with ctrl-+/-
- z/0 resets zoom
- adds mouse modes (no toolbar yet) with overrides (shift for select, space for drag, ctrl for time select)
- adds f for zooming in on a selected slice

### CODE REVIEW

File: gapic/src/main/com/google/gapid/perfetto/TraceView.java [hunk 1]
```diff
@@ -97,35 +97,78 @@ public TraceView(Composite parent, Models models, Widgets widgets) {
...
-        redraw = rootPanel.zoom(mouse.x, 1.0 + (fast ? KB_ZOOM_FAST : KB_ZOOM_SLOW)) || redraw;
+        redraw = rootPanel.zoom(mouse.x, 1.0 + (shift ? KB_ZOOM_FAST : KB_ZOOM_SLOW)) || redraw;
       }
 
-      if (kb.isKeyDown('m')) {
-        Selection selection = state.getSelection();
-        if (selection != null) {
-          selection.mark(state);
-          redraw = true;
+      rootPanel.setPanOverride(kb.isKeyDown(SWT.SPACE));
```
[the remark this answers is not in the document]
Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/perfetto/TraceView.java]: This callback is called as long as at least one key is down and once more, when no more keys are down. `kb.isKeyDown` returns false as soon as you let go of the space bar.

There is an issue with it on Windows when multiple keys are pressed. I have a fix for it, but it depends on changes in here.

File: gapic/src/main/com/google/gapid/perfetto/views/RootPanel.java [hunk 2]
```diff
@@ -170,32 +174,108 @@ public void visit(Visitor v, Area area) {
...
+      return Dragger.NONE;
+    }
+
+    MouseMode mode = mouseMode;
+    if ((mods & SWT.SHIFT) == SWT.SHIFT) {
+      mode = MouseMode.Select;
+    } else if ((mods & SWT.MOD1) == SWT.MOD1) {
+      mode = MouseMode.TimeSelect;
+    } else if (panOverride) {
+      mode = MouseMode.Pan;
```
[the remark this answers is not in the document]
Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/perfetto/views/RootPanel.java]: No, I want shift and control to have precedence.

File: gapic/src/main/com/google/gapid/perfetto/TraceView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (stellama0208) [reply gapic/src/main/com/google/gapid/perfetto/TraceView.java]: Somehow when I was testing on my glinux, the panOverride is not reset to false after I let go of the space bar. 
The reproduce procedure is: 
1. press '1'    -> go into selection mode
2. hold space for a while and release
3. try dragging     -> feels like pan mode (print out values will show it's in selection mode but have panOverride still being true)
4.press any key like 'a' and then drag again  -> go into selection mode, which is a little confusing

[the remark this answers is not in the document]
Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/perfetto/TraceView.java]: You are absolutely right! The "no more keys are down" event got lost. I've added a commit that should fix this. Thanks for testing it out thoroughly!

[the remark this answers is not in the document]
Comment (stellama0208) [reply gapic/src/main/com/google/gapid/perfetto/TraceView.java]: Np! It's my pleasure!

**Pull request merged**